code_samples/register_actions/src/org/jetbrains/tutorials/actions/SimplePopDialogAction.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+ */
+
+package org.jetbrains.tutorials.actions;
+
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.Messages;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+
+/**
+ * Action class to demonstrate how to interact with the IntelliJ Platform.
+ * The only action this class performs is to provide the user with a popup dialog as feedback.
+ * Typically this class is instantiated by the IntelliJ Platform framework based on declarations
+ * in the plugin.xml file. But when added at runtime this class is instantiated by an action group.
+ * @see com.intellij.openapi.actionSystem.AnAction
+ * @see com.intellij.openapi.actionSystem.AnActionEvent
+ */
+public class SimplePopDialogAction extends AnAction {
+
+  // Capture the generic message icon for display in the dialog.
+  private static final Icon ourIcon = Messages.getInformationIcon();
```

Comment:
Having static fields in `AnAction`-derived classes is a great way to introduce memory leaks, so we should not show this in our sample code, even though this particular usage is harmless. You can simply inline this field.

Reply:
I will change the field.
I'd like to better understand why static fields might introduce memory leaks. Perhaps we can discuss off-line.